packages/dev/core/src/Debug/physicsViewer.ts

```diff
@@ -70,10 +70,32 @@ export class PhysicsViewer {
         this._utilityLayer.utilityLayerScene.autoClearDepthAndStencil = true;
     }
 
-    /** @internal */
+    /**
+     * Updates the debug meshes of the physics engine.
+     *
+     * This code is useful for synchronizing the debug meshes of the physics engine with the physics impostor and mesh.
+     * It checks if the impostor is disposed and if the plugin version is 1, then it syncs the mesh with the impostor.
+     * This ensures that the debug meshes are up to date with the physics engine.

```

Comment:
I think there are some details added by ChatGPT that we don't really need in a function doc, as they are relative to the implementation of the function itself (like the 2nd sentence here)... We run the risk of getting the doc out of sync with the code when the code is updated in the future, because people won't think to update the doc.

Same thing in all docs below.

Reply:
As developers, we have to keep code and comment sync. There is no guarantee they'll keep in sync but we have to do our best. When I read the generated comment, I put myself in the user shoes having a problem when using the method and think to myself: 'Can this comment help user debug'? If yes, then I keep it. I remove around 30% of generated text. It sure can be better.

Reply:
Okay, but until now, our documents didn't describe the internal details of what a function does.

For the sake of consistency, we (and external users who do PR too) should apply the same policy: should we now detail what a function does? ChatGPT is currently handy for generating this doc automatically, but when it's no longer free, I doubt many people will pay a subscription (I personally won't), so we'll be on our own to write the doc.

It can be difficult enough to give an adequate description of a function (think of non-native English speakers), it will be even more difficult if we have to give a more in-depth description.

Also:
> As developers, we have to keep code and comment sync

We know how this works in practice ;)

> There is no guarantee they'll keep in sync but we have to do our best.

The more comments there are, the more likely they are to be out of sync with the code.

Also, we should only document when it makes the most sense and not for self-documented code. In your PR, there are many small self-documented functions that don't need detailed comments because you understand what they do just by reading the code. I understand that these comments are intended for the user who reads the doc of the function and not the code of this function but I still wonder about the interest of such comments for such small functions.

The last argument is that it is not common practice to put detailed comments about the internal workings of a function in the doc of that function. At least, I've never seen such a usage in the projects I've worked with (but I admit that I haven't worked on many different projects).

As you probably understood, I'm not really fond of this change! Maybe the new physics architecture can be used as a testbed and we can see how it evolves in the future...

It would be nice to have inputs from other people as it can impact everyone in the team: @BabylonJS/core-team-microsoft

Reply:
I'm using this v2 plugin as a testbed, indeed. That's why I name them 'iteration' and not after functionalities. What to put in the comment is an interesting topic we should discuss in the team.